.circleci/Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM    circleci/openjdk:latest
+
+# Based on https://github.com/GoogleCloudPlatform/cloud-builders/blob/master/bazel/Dockerfile
+RUN \
+    sudo sh -c 'echo "deb [arch=amd64] http://storage.googleapis.com/bazel-apt stable jdk1.8" > /etc/apt/sources.list.d/bazel.list '
+
+RUN \
+    curl https://storage.googleapis.com/bazel-apt/doc/apt-key.pub.gpg | sudo apt-key add -
+
+RUN sudo apt-get update && \
+    sudo apt-get -y install bazel \
+    wget software-properties-common make cmake python python-pip \
+    bc libtool automake zip time g++-6 gcc-6 \
+    rsync golang
+
```

Comment:
Can you reuse the dockerfile in istio/istio circleci/ ? We can have one big fat docker image that is pushed to docker hub.

Reply:
I don't think it's the best option - fat is also slow and complex. Modular 'micro-services' like images seem better. For example the istio/istio is based on golang - doesn't need jdk and bazel or gcc. I'm working  on another image to build on ARM - which needs arm and android cross compilers and cmake, but don't need java or bazel or go.

If you add the cross compilers - it's getting too fat. 

Even for istio/istio - I think one image to build with bazel and generate artifacts, and one with only go for the normal builds is cleaner.

Reply:
About gcloud - it's a leftover, I agree it should be under dockerhub.

But I was not sure about naming - istio/ci:bazel0.8-... ?  Or should we use 
istio/ci-bazel, etc ?

Reply:
Envoy’s build image is about 1.3GB. Circleci caches the images. And since these images don’t change often, it’s often very helpful for people to download a single build container and build all they want from there, just like the repo tool concept. Internal build systems can replicate the setup from a single dockerfile and go about work. 

We use the same build for api and c++ in Envoy as an example. 

While I am up for the smaller images, I would like us to first get quickly to a streamlined build with minimal overhead. Once we have a well oiled pipeline, we can split up images if we find things to be slow.